src/test/java/org/prebid/server/bidder/gothamads/GothamAdsBidderTest.java

```diff
@@ -0,0 +1,246 @@
+import static org.prebid.server.util.HttpUtil.USER_AGENT_HEADER;
+import static org.prebid.server.util.HttpUtil.X_FORWARDED_FOR_HEADER;
+import static org.prebid.server.util.HttpUtil.X_OPENRTB_VERSION_HEADER;
+import static org.prebid.server.util.HttpUtil.headers;
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE;
+
+public class GothamAdsBidderTest extends VertxTest {
+
+    private static final String ENDPOINT_URL = "https://test-url.com/?pass={{AccountId}}";
+    private static final String OPEN_RTB_VERSION = "2.5";
+    private static final String DEVICE_UA = "ua";
+    private static final String IPV_6_HEADER_VALUE = "ipv6";
+    private static final String IP_HEADER_VALUE = "ip";
+    private static final String IMP_ID = "imp_id";
+
+    private GothamAdsBidder sut;
+
+    @BeforeEach
+    public void setUp() {
+        sut = new GothamAdsBidder(ENDPOINT_URL, jacksonMapper);
+    }
+
+    @Test
+    public void shouldFailOnBidderCreation() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new GothamAdsBidder("invalid_url", jacksonMapper));
+    }
+
+    @Test
+    public void shouldReturnErrorWhileMakingHttpRequestsWhenRequestHasInvalidImpression() {
+        // given
+        final ObjectNode invalidExt = mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode()));
+        final BidRequest bidRequest = givenBidRequest(impBuilder -> impBuilder.ext(invalidExt));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = sut.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1).first()
+                .satisfies(error -> assertThat(error.getType()).isEqualTo(Type.bad_input));
+    }
+
+    @Test
+    public void shouldMakeCorrectRequest() {
+        // given
+        final String secondImpId = "imp_id2";
+        final BidRequest bidRequest = givenBidRequest(identity(), i -> i.id(secondImpId).ext(givenExt()));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = sut.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        final BidRequest expectedBidRequest = givenBidRequest(i -> i.ext(null), i -> i.id(secondImpId).ext(givenExt()));
+        assertThat(result.getValue()).usingRecursiveComparison()
+                .isEqualTo(expectedResults(expectedBidRequest, IMP_ID, secondImpId).getValue());
+    }
+
+    @Test
+    public void shouldReturnErrorWhileMakingBidsWhenResponseCanNotBeParsed() {
+        // given
+        final BidderCall<BidRequest> httpCall = givenHttpCall(null, "invalid");
+
+        // when
+        final Result<List<BidderBid>> actual = sut.makeBids(httpCall, null);
+
+        // then
+        assertThat(actual.getValue()).isEmpty();
+        assertThat(actual.getErrors()).containsExactly(badServerResponse("Bad Server Response"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("responsesWithoutSeatBids")
```

Comment:
Please, unwrap to separate unit tests. We tend not to use parametrized tests in unit tests as its not clear what are input parameters.

Reply:
it's clear, please check the test method params